nwb_conversion_tools/utils/si090nwbephyswriter.py

```diff
@@ -0,0 +1,199 @@
+from .common_writer_tools import PathType, default_return
+
+try:
+    import spikeinterface as si
+    from spikeinterface.core.testing_tools import generate_recording, generate_sorting
+
+    if StrictVersion(si.__version__[:4]) >= StrictVersion("0.90"):
+        HAVE_SI_090 = True
+    else:
+        HAVE_SI_090 = False
+except ImportError:
+    HAVE_SI_090 = False
+
+
+class SI090NwbEphysWriter(BaseSINwbEphysWriter):
+    """
+    Class to write a Recording, Sorting, Event, or WaveformExtractor object from SI>=0.90 to NWB
+
+    Parameters
+    ----------
+    object_to_write: si.BaseRecording, si.BaseSorting, si.BaseEvent, si.WaveformExtractor
+    nwb_file_path: Path type
+    nwbfile: pynwb.NWBFile or None
+    metadata: dict or None
+    **kwargs: list kwargs and meaning
+    """
+
+    def __init__(self, object_to_write, stub=False, stub_channels=None):
+        assert HAVE_SI_090
+        BaseSINwbEphysWriter.__init__(self, object_to_write, stub=stub, stub_channels=stub_channels)
+        if isinstance(self.object_to_write, si.BaseRecording):
+            self.recording = self.object_to_write
+            if self.stub:
+                self._make_recording_stub()
+        elif isinstance(self.object_to_write, si.BaseSorting):
+            self.sorting = self.object_to_write
+            if self.stub:
+                self._make_sorting_stub()
+        elif isinstance(self.object_to_write, si.BaseEvent):
+            self.event = self.object_to_write
+        elif isinstance(self.object_to_write, si.WaveformExtractor):
+            self.recording = self.object_to_write.recording
+            self.sorting = self.object_to_write.sorting
+            self.waveforms = self.object_to_write
+            if self.stub:
+                self._make_sorting_stub()
+                self._make_recording_stub()
+
+    def _make_recording_stub(self):
+        if self.stub_channels is not None:
+            channel_stub = self.stub_channels
+        else:
+            num_channels = min(10, self.recording.get_num_channels())
+            channel_stub = self.recording.get_channel_ids()[:num_channels]
+        frame_stub = min(100, self.recording.get_num_frames())
+        self.recording = si.ChannelSliceRecording(self.recording, channel_ids=channel_stub)
+        self.recording = si.FrameSliceRecording(self.recording, end_frame=frame_stub)
+
+    def _make_sorting_stub(self):
+        max_min_spike_time = max(
+            [min(x) for y in self.sorting.get_unit_ids() for x in [self.sorting.get_unit_spike_train(y)] if any(x)]
+        )
+        self.sorting = si.FrameSliceSorting(start_frame=0, end_frame=1.1 * max_min_spike_time)
+
+    @staticmethod
+    def supported_types():
+        assert HAVE_SI_090
+        return (si.BaseRecording, si.BaseSorting, si.BaseEvent, si.WaveformExtractor)
+
+    def get_num_segments(self):
+        return self.object_to_write.get_num_segments()
+
+    @default_return(None)
```

Comment:
I would remove all decorators for now. Again, I think that if something is wrong an exception shoudl be raised, and these decorators might catch the exception and make it hard to debug. @bendichter @CodyCBakerPhD opinions?

Reply:
the use of this decorator is to neatly deal with cases:
1. when the method returns a wrong dtype (list, vs array vs none); we can ensure the return value is of a specific dtype only. 
2. if the `self.recording`, `self.sorting` objects are none and the user calls the methods that require it, example calling `self._get_unit_spike_train_ids()` is called when the `object_to_write` is a `self.recording` (so `self.sorting` is None). Yes, calling spiketrainids is invalid in this case, but I think we should not throw an error but just return None.

Reply:
In my opinion we don't need them. They are only for internal methods (e.g. `_get_traces()`) so they are not supposed to be called by external users.